clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -1295,7 +1295,7 @@ public int compare(FetchResponse.AbortedTransaction o1, FetchResponse.AbortedTra
         }
 
         private boolean containsAbortMarker(RecordBatch batch) {
-            if (!batch.isControlBatch())
+            if (!(batch.isControlBatch() && batch.iterator().hasNext()))
```

Comment:
nit: seems like a better practice to use the same iterator instance for both `hasNext` and `next`. The construction of the iterator does do a little work. Same thing for the other checks.

Also, the `hasNext` check below is now redundant.